src/Workspaces/Remote/ServiceHub/Host/SolutionAssetCache.cs

```diff
@@ -218,22 +218,7 @@ private async ValueTask AddPinnedChecksumsAsync(HashSet<Checksum> pinnedChecksum
                 return;
 
             var checksums = await _remoteWorkspace.CurrentSolution.State.GetStateChecksumsAsync(cancellationToken).ConfigureAwait(false);
-
-            Recurse(checksums);
-
-            return;
-
-            void Recurse(ChecksumWithChildren checksums)
-            {
-                pinnedChecksums.Add(checksums.Checksum);
-                foreach (var child in checksums.Children)
-                {
-                    if (child is ChecksumWithChildren childChecksums)
```

Comment:
so many of these cases were not necessary.  they implied a recursive nature to checksums taht just does not exist.  specifically, a ChecksumWithChildren in the past itself never contained another ChecksumWithChildren.  So this was all pointless.  Removing this inheritance now makes everything clear as to what the relationship is.